### PR TITLE
Feature/seeds flag results

### PR DESCRIPTION
**Pull request by albrja**
### Description
Added a disaggregate_seeds flag (-d, --disaggregate-seeds) to make_results tool.
- *Category*: Feature
- *JIRA issue*: [MIC-3235](https://jira.ihme.washington.edu/browse/MIC-3235)
- *Research reference*: Request from research team.

-added flag in cli.py with default as False to preserve traditional make_results behavior.
-Updated build_results function call with logic for new flag and passed along bool argument to make_measure data.
-Updated make_measure_data and subsequent calls with disaggregate_seeds bool argument to handle new flag and include random seed column in results if flag is thrown.

### Verification and Testing
Successfully ran make_results both with and without the new disaggregate_seeds flag with the random seed column only appearing when expected.

### CODE REVIEW

File: src/vivarium_gates_iv_iron/results_processing/process_results.py [hunk 1]
```diff
@@ -142,64 +142,78 @@ def aggregate_over_seed(data):
...
+    data = data[sort_order + other_cols + ['value']].sort_values(sort_order)
     return data.reset_index(drop=True)
 
 
-def apply_results_map(data, kind):
+def apply_results_map(data: pd.DataFrame, kind: str) -> pd.DataFrame:
     logger.info(f"Mapping {kind} data to stratifications.")
     map_df = results.RESULTS_MAP(kind)
-    data = data.set_index("key")
+    data = data.set_index('key')
```
Comment (mattkappel) [inline src/vivarium_gates_iv_iron/results_processing/process_results.py]: Should be double-quotes for strings.

File: src/vivarium_gates_iv_iron/results_processing/process_results.py [hunk 2]
```diff
@@ -142,64 +142,78 @@ def aggregate_over_seed(data):
...
-def get_transition_count_measure_data(data, measure):
+def get_transition_count_measure_data(data: pd.DataFrame, measure: str, disaggregate_seeds: bool) -> pd.DataFrame:
     # Oops, edge case.
-    data = data.drop(
-        columns=[c for c in data.columns if "event_count" in c and "2041" in c]
-    )
-    data = get_measure_data(data, measure)
-    return sort_data(data)
+    data = data.drop(columns=[c for c in data.columns if 'event_count' in c and str(results.YEARS[-1]+1) in c])
+    data = get_measure_data(data, measure, disaggregate_seeds)
```
Comment (mattkappel) [inline src/vivarium_gates_iv_iron/results_processing/process_results.py:249]: Formatting nitpick -- please add spaces around the '+'.

Also, `"event_count"`

File: src/vivarium_gates_iv_iron/results_processing/process_results.py [hunk 3]
```diff
@@ -142,64 +142,78 @@ def aggregate_over_seed(data):
...
+    if disaggregate_seeds:
+        groupby_cols = GROUPBY_COLUMNS + [results.RANDOM_SEED_COLUMN]
+    else:
+        groupby_cols = GROUPBY_COLUMNS
 
+    total_pop = pivot_data(data[[results.TOTAL_POPULATION_COLUMN]
+                                + results.RESULT_COLUMNS('population')
+                                + groupby_cols], disaggregate_seeds)
+    total_pop = total_pop.rename(columns={'key': 'measure'})
+    return sort_data(total_pop, disaggregate_seeds)
```
Comment (mattkappel) [inline src/vivarium_gates_iv_iron/results_processing/process_results.py:201]: Formatting nitpick 😄

**Review by mattkappel — APPROVED** [review]
Looks good, aside from some PEP-8 regressions!

**Review by mattkappel — APPROVED** [review]

**Review by rmudambi — APPROVED** [review]

**Pull request merged**